clippy_lints/src/methods/mod.rs

```diff
@@ -2367,14 +2367,18 @@ fn lint_search_is_some<'a, 'tcx>(
         let search_snippet = snippet(cx, search_args[1].span, "..");
         if search_snippet.lines().count() <= 1 {
             // suggest `any(|x| ..)` instead of `any(|&x| ..)` for `find(|&x| ..).is_some()`
+            // suggest `any(|..| *..)` instead of `any(|..| **..)` for `find(|..| **..).is_some()`
             let any_search_snippet = if_chain! {
                 if search_method == "find";
                 if let hir::ExprKind::Closure(_, _, body_id, ..) = search_args[1].node;
                 let closure_body = cx.tcx.hir().body(body_id);
                 if let Some(closure_arg) = closure_body.arguments.get(0);
-                if let hir::PatKind::Ref(..) = closure_arg.pat.node;
                 then {
-                    Some(search_snippet.replacen('&', "", 1))
+                    if let hir::PatKind::Ref(..) = closure_arg.pat.node {
+                        Some(search_snippet.replacen('&', "", 1).replacen("**", "*", 1))
+                    } else {
+                        Some(search_snippet.replacen("**", "*", 1))
+                    }
```

Comment:
IIUC, the `search_snippet` is the whole closure in the `find` call. The `replacen` for the `|&_|` only works, because the closure is guaranteed to start with `|&_|`.

The deref has 3 problems: 
1. Not only the closure arg can be derefed: `(0..1).find(|x| **y == *x)`
2. It may be derefed only once or multiple times: `(0..1).find(|x| *x == 0).is_some()`
3. When the closure arg already has `PatKind::Ref`, then the amount of derefs must stay the same. (This is the easy case)

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=dbf6bc1a804768b1d2e1aa3e12feaba8) for the three examples. (Please also add these as tests, if they don't exist yet.)

So to answer your question:

> do you think this is enough, or is some logic required?

There is some logic required. Since we currently only emit the suggestion, when the closure is a one-liner, I think it is save to say, that the closure arguments doesn't get shadowed. This means, you have to get the `Ident` of the closure arg and then replace every occurrence of `*<ident-name>` with `<ident-name>`.

---
To do it right, and also accounting for shadowing of the closure arg, we would have to write a visitor, that searches for occurrences of `Deref(<closure_arg>)` and then only replace these derefs. But I **really** don't think, that this is necessary.